src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -787,8 +787,8 @@ private void handleOne(final Callback[] callbacks, final int idx) throws IOExcep
                     handleOne(callbacks, idx + 1);
                 } else if (callback instanceof PasswordCallback) {
                     final PasswordCallback passwordCallback = (PasswordCallback) callback;
-
-                    if (getCredentialAcquireSupport(PasswordCredential.class).mayBeSupported()) {
+                    SupportLevel supportLevel = getCredentialAcquireSupport(PasswordCredential.class);
```

Comment:
I think was a false issue in the static analysis.

A RealmIdentity should not be returning null for SupportLevel - a null SupportLevel has no meaning,  Instead I think the javadoc for RealmIdentity should be checked to ensure it is clear null must not be returned.  

If we want to be complete maybe the state that calls the RealmIdentity should double check null was not returned and throw an appropriate exception.